src/Bicep.Core/Parser/Parser.cs

```diff
@@ -773,7 +783,7 @@ current.Type switch
                 var value = this.WithRecovery(() => Expression(allowComplexLiterals: true), GetSuppressionFlag(colon), TokenType.NewLine);
 
                 return new ObjectPropertySyntax(key, colon, value);
-            }, RecoveryFlags.None, TokenType.NewLine);
+            }, RecoveryFlags.None, TokenType.NewLine, TokenType.Comma);
```

Comment:
Is this to handle the case where we've got some invalid syntax as well as a comma? If so, I think it would be better to avoid complicating the recovery logic, and just let the original error bubble up.

Reply:
You are correct, made the change